### ISSUE TITLE
smart quotes getting wrong in latex (at least with list of strings via autoattribute)

### ISSUE THREAD

**Issue by jfbu** [sphinx-doc/sphinx#3472]
Subject: when compiling Sphinx's own docs to LaTeX, one finds this mark-up 

    = {[}'image/svg+xml', `image/png', `image/gif', `image/jpeg'{]}

in the Builders section. This comes from an 

       .. autoattribute:: supported_image_types

and the class variable in the Python code is 

        supported_image_types = ['image/svg+xml', 'image/png',
                                 'image/gif', 'image/jpeg']

The first quote in the LaTeX output is wrong, it should be a left quote. This does not show up so much with the used fonts for the doc, but if one changes fonts it is clearer:

![capture d ecran 2017-02-26 a 23 11 20](https://cloud.githubusercontent.com/assets/2589111/23344286/00557c42-fc7a-11e6-8a0b-94d11c7f6200.png)

HTML build uses (right) tick both on opening and closing for all strings ``'``.

### Environment info
- OS: mac os x 10.9.5
- Python version: 3.5.2
- Sphinx version: 1.5.2, also on master branch.

**Comment by tk0miya** [comment]
It might be related with `sphinx.util.smartypants`.
It provides following regexp.
```
# Get most opening single quotes:
opening_single_quotes_regex = re.compile(r"""
                (
                        \s          |   # a whitespace char, or
                        &#160;      |   # a non-breaking space entity, or
                        --          |   # dashes, or
                        &[mn]dash;  |   # named dash entities
                        %s          |   # or decimal entities
                        &\#x201[34];    # or hex
                )
                '                 # the quote
                (?=\w)            # followed by a word character
                """ % (dec_dashes,), re.VERBOSE)
```

**Comment by bfrk** [comment]
I have a problem here with syntax descriptions that might be related. For instance, the keyword "entry" in

```
.. productionlist::
   entry: "entry" `block`
```

gets rendered as

```
``entry''
```

in the latex output.

**Comment by tk0miya** [comment]
@bfrk it might be a bug of smartypants processing.
It is similar to this issue, but it is not related problem.
So please file a new issue.

Thank you for reporting

**Comment by jfbu** [comment]
@tk0miya the new issue was filed as #3507, it is unrelated to this one.

**Comment by jfbu** [comment]
Closing as this was fixed at 1.6.1 from merge  of #3666.
